stdlib/3/array.pyi

```diff
@@ -2,48 +2,58 @@
 
 # Based on http://docs.python.org/3.2/library/array.html
 
-from typing import Any, Iterable, Tuple, List, Iterator, BinaryIO, overload
+from typing import (Any, BinaryIO, Generic, Iterable, Iterator, List, MutableSequence,
+                    overload, Text, Tuple, TypeVar, Union)
+
+_T = TypeVar('_T', int, float, Text)
 
 typecodes = ...  # type: str
 
-class array:
+class array(MutableSequence[_T], Generic[_T]):
     typecode = ...  # type: str
     itemsize = ...  # type: int
     def __init__(self, typecode: str,
-                 initializer: Iterable[Any] = ...) -> None: ...
-    def append(self, x: Any) -> None: ...
+                 __initializer: Union[bytes, Iterable[_T]] = ...) -> None: ...
+    def append(self, x: _T) -> None: ...
     def buffer_info(self) -> Tuple[int, int]: ...
     def byteswap(self) -> None: ...
     def count(self, x: Any) -> int: ...
-    def extend(self, iterable: Iterable[Any]) -> None: ...
+    def extend(self, iterable: Iterable[_T]) -> None: ...
     def frombytes(self, s: bytes) -> None: ...
     def fromfile(self, f: BinaryIO, n: int) -> None: ...
-    def fromlist(self, list: List[Any]) -> None: ...
+    def fromlist(self, list: List[_T]) -> None: ...
     def fromstring(self, s: bytes) -> None: ...
     def fromunicode(self, s: str) -> None: ...
-    def index(self, x: Any) -> int: ...
-    def insert(self, i: int, x: Any) -> None: ...
-    def pop(self, i: int = ...) -> Any: ...
+    def index(self, x: _T) -> int: ...  # type: ignore  # Overrides Sequence
+    def insert(self, i: int, x: _T) -> None: ...
+    def pop(self, i: int = ...) -> _T: ...
     def remove(self, x: Any) -> None: ...
     def reverse(self) -> None: ...
     def tobytes(self) -> bytes: ...
     def tofile(self, f: BinaryIO) -> None: ...
-    def tolist(self) -> List[Any]: ...
+    def tolist(self) -> List[_T]: ...
     def tostring(self) -> bytes: ...
     def tounicode(self) -> str: ...
 
     def __len__(self) -> int: ...
-    def __iter__(self) -> Iterator[Any]: ...
-    def __str__(self) -> str: ...
-    def __hash__(self) -> int: ...
 
     @overload
-    def __getitem__(self, i: int) -> Any: ...
+    def __getitem__(self, i: int) -> _T: ...
+    @overload
+    def __getitem__(self, s: slice) -> 'array[_T]': ...
+
+    @overload  # type: ignore  # Overrides MutableSequence
```

Comment:
`MutableSequence`, the supertype, defines `__setitem__()` like this:
```python
@overload
def __setitem__(self, i: int, o: _T) -> None: ...
@overload
def __setitem__(self, s: slice, o: Iterable[_T]) -> None: ...
```
But `array` requires an `array` instance. The `# type: ignore` comment quiets the `Signature of "__setitem__" incompatible with supertype "MutableSequence"` error emitted by mypy, but pytype doesn't emit an error when the type comment is omitted nor does it like the type comment.

Any thoughts on how to handle this one?

Reply:
I can't think of anything else here than lying about the signature and putting `Iterable[_T]`. We can put a TODO to fix it when pytype's parser becomes more permissive (cc @matthiaskramm).